src/components/common/CopyTooltip/index.tsx

```diff
@@ -32,24 +32,69 @@ const CopyTooltip = ({
         setShowConfirmation(true)
         return
       }
+
       let timeout: NodeJS.Timeout | undefined
 
-      try {
-        navigator.clipboard.writeText(text).then(() => setTooltipText('Copied'))
-        setShowConfirmation(false)
-        setShowTooltip(true)
-        timeout = setTimeout(() => {
-          if (isCopyEnabled) {
-            setShowTooltip(false)
-            setTooltipText(initialToolTipText)
+      const copyToClipboard = async (text: string): Promise<boolean> => {
+        if (navigator.clipboard && window.isSecureContext) {
+          // Use Clipboard API if available and in secure context
+          try {
+            await navigator.clipboard.writeText(text)
+            return true
+          } catch (err) {
+            console.warn('Clipboard API failed:', err)
           }
-        }, 750)
-        onCopy?.()
-      } catch (err) {
-        setIsCopyEnabled(false)
-        setTooltipText('Copying is disabled in your browser')
+        }
+
+        // Fallback for iOS
+        const textArea = document.createElement('textarea')
+        textArea.value = text
+        // Avoid scrolling to bottom
+        textArea.style.top = '0'
+        textArea.style.left = '0'
+        textArea.style.position = 'fixed'
+        document.body.appendChild(textArea)
+        textArea.focus()
+        textArea.select()
+
+        try {
+          // Using touch events for iOS
+          const event = new TouchEvent('touch')
+          textArea.dispatchEvent(event)
+          document.execCommand('selectall')
+          const successful = document.execCommand('copy')
```

Comment:
This method is deprecated: https://developer.mozilla.org/en-US/docs/Web/API/Document/execCommand

I would rather not add this fallback. Any idea why navigator.clipboard might not work on iOS?

Reply:
> Secure context: This feature is available only in [secure contexts](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts) (HTTPS), in some or all [supporting browsers](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/clipboard#browser_compatibility).

Is it possible that we are testing on non-https page?

Reply:
> > Secure context: This feature is available only in [secure contexts](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts) (HTTPS), in some or all [supporting browsers](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/clipboard#browser_compatibility).
> 
> Is it possible that we are testing on non-https page?
Yeah, it works on https pages, but not on localhost and non-https pages

Reply:
> This method is deprecated: https://developer.mozilla.org/en-US/docs/Web/API/Document/execCommand
> 
> I would rather not add this fallback. Any idea why navigator.clipboard might not work on iOS?

`navigator.clipboard` is working on https pages only not on localhost.

Reply:
Then I suggest we just log a warning if clipboard fails but not introduce a fallback just for dev environments. Thanks for figuring out the root cause @HarshS1611!